packages/runtime/container-runtime/src/summary/summarizerTypes.ts

```diff
@@ -231,12 +237,12 @@ export interface IAckSummaryResult {
 	readonly ackNackDuration: number;
 }
 
-export interface INackSummaryResult {
+export interface INackSummaryResult extends ISubmitSummaryFailureResult {
 	readonly summaryNackOp: ISummaryNackMessage;
 	readonly ackNackDuration: number;
 }
 
-export type SummarizeResultPart<TSuccess, TFailure = undefined> =
+export type SummarizeResultPart<TSuccess, TFailure> =
```

Comment:
Can we say this?

```suggestion
export type SummarizeResultPart<TSuccess, TFailure extends IBaseSummaryFailureResult> =
```

Reply:
Actually, I'm not sure why we want `stage` on all three, it only applies to the SubmitSummary, right?

Reply:
You are right. It only applies to submitSummary. I simplified the whole logic, so this change is removed anyway.